src/java/org/apache/cassandra/index/sai/disk/v1/MemtableIndexWriter.java

```diff
@@ -83,13 +89,31 @@ public void addRow(PrimaryKey key, Row row, long sstableRowId)
     {
         // Memtable indexes are flushed directly to disk with the aid of a mapping between primary
         // keys and row IDs in the flushing SSTable. This writer, therefore, does nothing in
-        // response to the flushing of individual rows.
+        // response to the flushing of individual rows except for keeping index-specific statistics.
+        boolean isStatic = indexTermType.columnMetadata().isStatic();
+
+        // Indexes on static columns should only track static rows, and indexes on non-static columns 
+        // should only track non-static rows. (Within a partition, the row ID for a static row will always
+        // come before any non-static row.) 
+        if (key.kind() == PrimaryKey.Kind.STATIC && isStatic || key.kind() != PrimaryKey.Kind.STATIC && !isStatic)
+        {
+            if (minKey == null)
+                minKey = key;
+            maxKey = key;
```

Comment:
This way of updating min/max keys is based on the expectation that `PerColumnIndexWriter#addRow` is always called in primary key order. Maybe we can mention that in the JavaDoc for `PerColumnIndexWriter`?

Reply:
Yeah, we should make that explicit in the `PCIW` method doc.